### PR TITLE
Clean, expand and document JsonFields.

### PR DESCRIPTION
**Pull request by rec**
As part of my grand survey of existing RPC commands, I need to isolate all the data members of JSON input and output to rippled.

This is a pretty huge task, and I imagine there are still a few missing, but I scoured the code and am reasonably confident of having found nearly all of them.

Reviewers: @nbougalis @josh-ripple

### CODE REVIEW

**Comment by scottschurr** [comment]
I've just taken a quick gander at this code.  I like like the idea and appreciate the effort.

In my work with TransactionSign.cpp, I've sometimes been using the SField.fieldName member to try and reduce the wide distribution of random quoted strings.  I wonder if jss competes with SField for capturing some of these strings.  If so, is it worth dealing with that?  Not saying it is, just wondering.

File: src/ripple/protocol/JsonFields.h [hunk 1]
```diff
@@ -34,153 +34,346 @@ namespace jss {
...
+JSS ( account_hash );               // out: LedgerToJson
+JSS ( account_id );                 // out: WalletPropose
+JSS ( account_index );              // in: AccountCurrencies, AccountOffers,
+                                    //     AccountInfo, AccountLines, OwnerInfo
+                                    // out: AccountOffers
+JSS ( account_root );               // in: LedgerEntry
+JSS ( accounts );                   // in: LedgerEntry, Subscribe,
+                                    //     handlers/Ledger, Unsubscribe
+                                    // out: WalletAccounts
+JSS ( accounts_proposed );          // in: Subscribe, Unsubscribe
```
Comment (sublimator) [inline src/ripple/protocol/JsonFields.h]: zomg, a MACRO

**Comment by rec** [comment]
Yes, there is definitely duplication between these and fields.  Indeed, one
of the category of comments I have is "field", which means that this is a
field.

And I'll bet I need to go and check to see I got all of them!  (because I
did them one at a time and I never went systematically through SFFields at
the end...)

back...  too many to count.

actually, I think the whole idea of having collisions between the SFields
and the StaticStrings is a bad one.  I think that a field should *either* be
right from the database, or it should have a completely different name...

**Comment by rec** [comment]
I realize that no matter when I check this in, there will probably be huge conflicts unless I do it soonest.

Can I get approval so I can push it before the weekend pls?

**Comment by rec** [comment]
(I booted Travis...)

**Comment by rec** [comment]
Travis is happy now.  Happy Travis!

**Comment by rec** [comment]
Ping!  @nbougalis @josh-ripple

**Comment by rec** [comment]
C'mon folks - this should be uncontroversial, take a look!

**Comment by nbougalis** [comment]
This is uncontroversial (modulo that one typo I spotted and left a comment for). I have no qualms in giving it a :+1: ahead of the typo fix.

**Comment by josh-ripple** [comment]
:+1:

**Pull request closed without merging**